File: testdata/bench/rust/BenchmarkTest00004.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: SQL injection prevented with sqlx::query! macro
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("name").unwrap_or(&String::new()).clone();
    // sqlx::query!("SELECT * FROM users WHERE name = $1", param).fetch_one(&pool).await
    HttpResponse::Ok().body("safe")
}

File: testdata/bench/javascript/BenchmarkTest00004.js
// Safe: SQL injection prevented via ORM (Knex query builder)
const express = require('express');
const knex = require('./knex');
const router = express.Router();
router.get('/test004', (req, res) => {
    const param = req.query.input;
    knex('users').where('id', param).first()
        .then(user => res.json(user))
        .catch(err => res.status(500).json({ error: 'Database error' }));
});
module.exports = router;
